### DISCUSSION TITLE
Is it possible to use this code with the emulator library?

### DISCUSSION THREAD

**Discussion by jwestes** [RudolphRiedel/FT800-FT813#74] [Q&A]
If this was possible, that would make development of a UI so much faster!

**Comment by RudolphRiedel** [comment]
What "emulator library"?
I started a couple of times to add an exporter to EVE Screen Editor, but I have to admit that Python is not exactly my cup of tea.
And I never figured out how to use EVE Screeen Designer even remotely productive.

Well, ESE exports something like this for EVE_HAL2:
Gpu_CoCmd_Text(phost, 108, 66, 27, OPT_CENTERX, "Lala 1");
Gpu_CoCmd_Text(phost, 208, 66, 27, OPT_CENTERX, "Lala 2");
Gpu_CoCmd_Text(phost, 308, 66, 27, OPT_CENTERX, "Lala 3");
Gpu_CoCmd_Text(phost, 408, 66, 27, OPT_CENTERX, "Lala 4");
Gpu_CoCmd_Slider(phost, 100, 100, 16, 300, 0, 30, 100);
Gpu_CoCmd_Slider(phost, 200, 100, 16, 300, 0, 30, 100);
Gpu_CoCmd_Slider(phost, 300, 100, 16, 300, 0, 30, 100);
Gpu_CoCmd_Slider(phost, 400, 100, 16, 300, 0, 30, 100);

And my library would need it like this:
EVE_cmd_text(108, 66, 27, OPT_CENTERX, "Lala 1");
EVE_cmd_text(208, 66, 27, OPT_CENTERX, "Lala 2");
EVE_cmd_text(308, 66, 27, OPT_CENTERX, "Lala 3");
EVE_cmd_text(408, 66, 27, OPT_CENTERX, "Lala 4");
EVE_cmd_slider(100, 100, 16, 300, 0, 30, 100);
EVE_cmd_slider(200, 100, 16, 300, 0, 30, 100);
EVE_cmd_slider( 300, 100, 16, 300, 0, 30, 100);
EVE_cmd_slider(400, 100, 16, 300, 0, 30, 100);

So for the most part this already is copy-paste now since I implemented the API as described in the programming guide.
But yes, I agree that direct ESE export might make some designs easier.

**Comment by jwestes** [comment]
There is an emulator that comes with the Screen Editor and FTDI provides documentation for how it works.  I've not looked at it much, so I don't know what it would take to make this work.
https://brtchip.com/wp-content/uploads/Support/Documentation/Programming_Guides/ICs/EVE/AN_281_FT800_Emulator_Library_User_Guide.pdf

**Reply by RudolphRiedel** [reply]
I am not sure what this one is good for since  ESE already is displaying the display content and ESE could also directly connect to a display using one of the USB 2 SPI adapters. But it does not seem to be working with ESE 4.4.0, the emulator window is not showing my display content.
Anyways, this one is using one of the four (or three ?) different API versions from FTDI/BRT and ESE just knows how to feed it.

Most interesting would be using the "Export" function of ESE:
![grafik](https://user-images.githubusercontent.com/31180093/211170112-86454fdf-a360-435c-8613-3aed032c4a84.png)

But so far I did not get deep enough into this to produce anything remotely usefull.

**Comment by jwestes** [comment]
Oh boy, I was not thinking clearly on this one.  Let's just forget I asked.  I was picturing a way to test without having to compile and run it on the MC.  I obviously didn't think through it though.  I've spent too much time working in the desktop world. :)
